pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2223,24 +2223,34 @@ func appendListenerFallthroughRoute(l *xdsapi.Listener, opts *buildListenerOpts,
 		}
 	}
 
-	if currentListenerEntry != nil {
-		for _, fc := range currentListenerEntry.listener.FilterChains {
-			if isMatchAllFilterChain(fc) {
-				// We can only have one wildcard match. If the existing filter chain already has one, skip it
-				// This can happen when there are multiple https services
-				return
-			}
-		}
+	fallthroughNetworkFilters := buildOutboundCatchAllNetworkFiltersOnly(push, node)
+
+	in := &plugin.InputParams{
+		Node: node,
+		Push: push,
+	}
+	mutable := &istionetworking.MutableObjects{
+		FilterChains: []istionetworking.FilterChain{
+			{
+				TCP: fallthroughNetworkFilters,
+			},
+		},
 	}
 
-	fallthroughNetworkFilters := buildOutboundCatchAllNetworkFiltersOnly(opts.push, node)
+	for _, p := range configgen.Plugins {
+		err := p.OnOutboundPassthroughFilterChain(in, mutable)
+		if err != nil {
+			log.Debugf("fail to set pass through filter chain for listener: %#v", l.Name)
+			return
+		}
+	}
 
-	opts.filterChainOpts = append(opts.filterChainOpts, &filterChainOpts{
-		filterChainName: util.PassthroughFilterChain,
-		networkFilters:  fallthroughNetworkFilters,
-		isFallThrough:   true,
-	})
-	l.FilterChains = append(l.FilterChains, &listener.FilterChain{FilterChainMatch: wildcardMatch})
+	outboundPassThroughFilterChain := &listener.FilterChain{
+		FilterChainMatch: &listener.FilterChainMatch{},
+		Name:             util.PassthroughFilterChain,
+		Filters:          mutable.FilterChains[0].TCP,
+	}
```

Comment:
What if the listener has no filter chain matches to start with? wont there be two filter chains with no matches? esp for TCP services

Reply:
Good question! It is resolved in the follow up PR, which adds the dst port in all bind_to_port=false listener.
I don't know why the integration tests are passing... Let me verify

Reply:
Confirmed. The magic is that we don't build passthrough filter chain if there is an existing catch all filter chain.

```
func (configgen *ConfigGeneratorImpl) appendListenerFallthroughRouteForCompleteListener(l *xdsapi.Listener, node *model.Proxy, push *model.PushContext) {
	for _, fc := range l.FilterChains {
		if isMatchAllFilterChain(fc) {
                   return
                }
       }
...
}
```